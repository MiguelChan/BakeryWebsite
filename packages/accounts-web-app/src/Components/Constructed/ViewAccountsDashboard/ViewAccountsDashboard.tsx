import {
  Account,
  isNullOrUndefined,
} from '@mgl/shared-components';
import React from 'react';
import {
  useHistory,
  useLocation,
} from 'react-router-dom';
import { GetAccountsApiFn } from '../../../Clients';
import {
  AccountsAppContext,
  ApplicationContext,
} from '../../../Context';
import {
  useGetAccounts,
  UseGetAccounts,
  UseGetAccountsState,
} from '../../../Hooks';
import { AccountsTable } from '../../Composites';

export interface ViewAccountsDashboardProps {
}

export const ViewAccountsDashboard: React.FunctionComponent<ViewAccountsDashboardProps> = ({
}) => {
  const appDependencies: ApplicationContext = React.useContext<ApplicationContext>(AccountsAppContext);
  const useGetAccountState: UseGetAccountsState = appDependencies.useGetAccounts();
  const history = useHistory();

  const onAccountClickedListener = (account: Account): void => {
    const {
      id,
    } = account;

    history.push(`accounts/${id}`);
  };

  const renderComponent = (): React.ReactElement => {
    const {
      accounts,
      isLoading,
      errorMessage,
    } = useGetAccountState;

    if (isLoading) {
      return <span>Loading data...</span>;
    }

    if (!isNullOrUndefined(errorMessage)) {
      return <span>{errorMessage}</span>;
    }

    return (
      <AccountsTable
        accounts={accounts}
        onAccountClickedListener={onAccountClickedListener}
      />
    );
  };

  return renderComponent();
};
