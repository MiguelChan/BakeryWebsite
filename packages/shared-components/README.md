# `@mgl/shared-components`

> TODO: description

## Usage

```
const sharedComponents = require('@mgl/shared-components');

// TODO: DEMONSTRATE API
```
