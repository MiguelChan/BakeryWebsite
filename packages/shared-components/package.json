{
  "name": "@mgl/shared-components",
  "version": "1.1.0",
  "description": "The WebApp components library",
  "main": "./build/src/index.js",
  "types": "./build/src/index.d.ts",
  "files": [
    "./build/src/index.js"
  ],
  "scripts": {
    "clean": "rm -rf build",
    "build": "tsc",
    "test": "jest --config jest.config.js",
    "lint": "eslint . --ext .ts",
    "lint:fix": "eslint . --ext .ts --fix",
    "release": "npm run clean && npm run lint && npm run test && npm run build"
  },
  "devDependencies": {
    "eslint-plugin-modules-newline": "^0.0.6"
  },
  "peerDependencies": {}
}
