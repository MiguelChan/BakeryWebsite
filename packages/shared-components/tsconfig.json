{
    "compilerOptions": {
      "target": "ESNext",
      "module": "commonjs",
      "outDir": "./build",
      "declarationDir": "build",
      "declaration": true,
      "lib": ["ESNext"],
      "types": ["node", "jest"]
    },
    "include": [
      "src/**/*",
      "tst/**/*"
    ],
    "exclude": [
        "node_modules",
        "build"
    ]
  }