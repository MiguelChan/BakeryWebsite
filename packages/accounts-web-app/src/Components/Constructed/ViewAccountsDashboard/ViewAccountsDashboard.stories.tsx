import React from 'react';
import {
  Story,
  Meta,
} from '@storybook/react';
import { AccountType } from '@mgl/shared-components';
import {
  ViewAccountsDashboard,
  ViewAccountsDashboardProps,
} from './ViewAccountsDashboard';
import {
  AccountsAppContext,
  ApplicationContext,
} from '../../../Context';
import { UseGetAccountsState } from '../../../Hooks';

export default {
  title: 'Components/Constructed/ViewAccountsDashboard',
  component: ViewAccountsDashboard,
} as Meta;

const Template: Story<ViewAccountsDashboardProps & ApplicationContext> = (props) => (
  <AccountsAppContext.Provider value={{ ...props }}>
    <ViewAccountsDashboard />
  </AccountsAppContext.Provider>
);

export const LoadingTable = Template.bind({});
LoadingTable.args = {
  useGetAccounts: (): UseGetAccountsState => ({
    accounts: [],
    isLoading: true,
  }),
};

export const EmptyTable = Template.bind({});
EmptyTable.args = {
  useGetAccounts: (): UseGetAccountsState => ({
    accounts: [],
    isLoading: false,
  }),
};

export const WithData = Template.bind({});
WithData.args = {
  useGetAccounts: (): UseGetAccountsState => ({
    accounts: [
      {
        accountType: AccountType.Entry,
        id: 'SomeIdSOmeId',
        subAccounts: [],
        title: 'Una cuenta',
      },
    ],
    isLoading: false,
  }),
};

export const WithErrorMessages = Template.bind({});
WithErrorMessages.args = {
  useGetAccounts: (): UseGetAccountsState => ({
    accounts: [],
    isLoading: false,
    errorMessage: 'Something went wrong',
  }),
};
