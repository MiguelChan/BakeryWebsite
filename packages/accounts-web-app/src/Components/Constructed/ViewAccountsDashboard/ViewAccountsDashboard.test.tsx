import React from 'react';
import {
  render,
  RenderResult,
} from '@testing-library/react';
import { AccountType } from '@mgl/shared-components';
import {
  AccountsAppContext,
  ApplicationContext,
} from '../../../Context';
import { ViewAccountsDashboard } from './ViewAccountsDashboard';
import {
  UseCreateAccountState,
  UseGetAccountsState,
} from '../../../Hooks';

describe('ViewAccountsDashbard', () => {
  const setupComponent = (props: ApplicationContext): RenderResult => {
    const Component = (
      <AccountsAppContext.Provider value={props}>
        <ViewAccountsDashboard />
      </AccountsAppContext.Provider>
    );
    return render(Component);
  };

  it('Should display the Loading Message', () => {
    const loadingTableProps: ApplicationContext = {
      useGetAccounts: (): UseGetAccountsState => ({
        accounts: [],
        isLoading: true,
      }),
      useCreateAccount: (): UseCreateAccountState => ({
        isAccountCreated: false,
        isLoading: false,
        requestCreateAccount: (): void => {},
        errorMessage: null,
      }),
      useGetAccount: jest.fn(),
    };

    const result = setupComponent(loadingTableProps);

    expect(result.container).toMatchSnapshot();
  });

  it('Should display the error message', () => {
    const errorMessageTableProps: ApplicationContext = {
      useGetAccounts: (): UseGetAccountsState => ({
        accounts: [],
        isLoading: false,
        errorMessage: 'Something went wrong',
      }),
      useCreateAccount: (): UseCreateAccountState => ({
        isAccountCreated: false,
        isLoading: false,
        requestCreateAccount: (): void => {},
        errorMessage: null,
      }),
      useGetAccount: jest.fn(),
    };

    const result = setupComponent(errorMessageTableProps);

    expect(result.container).toMatchSnapshot();
  });

  it('Should display empty data', () => {
    const emptyDataProps: ApplicationContext = {
      useGetAccounts: (): UseGetAccountsState => ({
        accounts: [],
        isLoading: false,
        errorMessage: undefined,
      }),
      useCreateAccount: (): UseCreateAccountState => ({
        isAccountCreated: false,
        isLoading: false,
        requestCreateAccount: (): void => {},
        errorMessage: null,
      }),
      useGetAccount: jest.fn(),
    };

    const result = setupComponent(emptyDataProps);

    expect(result.container).toMatchSnapshot();
  });

  it('Should display the data', () => {
    const withDataProps: ApplicationContext = {
      useGetAccounts: (): UseGetAccountsState => ({
        accounts: [
          {
            accountType: AccountType.Entry,
            id: 'SomeId',
            subAccounts: [],
            title: 'A title',
          },
        ],
        isLoading: false,
        errorMessage: undefined,
      }),
      useCreateAccount: (): UseCreateAccountState => ({
        isAccountCreated: false,
        isLoading: false,
        requestCreateAccount: (): void => {},
        errorMessage: null,
      }),
      useGetAccount: jest.fn(),
    };

    const result = setupComponent(withDataProps);

    expect(result.container).toMatchSnapshot();
  });
});
