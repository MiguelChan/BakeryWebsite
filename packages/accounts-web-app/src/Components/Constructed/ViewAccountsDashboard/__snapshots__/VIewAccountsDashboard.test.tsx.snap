// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewAccountsDashbard Should display empty data 1`] = `
<div>
  <table
    class="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
  >
    <thead
      class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
          scope="col"
        >
          Id
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
          scope="col"
        >
          Nombre de cuenta
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
          scope="col"
        >
          Tipo de Cuenta
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    />
  </table>
</div>
`;

exports[`ViewAccountsDashbard Should display the Loading Message 1`] = `
<div>
  <span>
    Loading data...
  </span>
</div>
`;

exports[`ViewAccountsDashbard Should display the data 1`] = `
<div>
  <table
    class="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
  >
    <thead
      class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
          scope="col"
        >
          Id
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
          scope="col"
        >
          Nombre de cuenta
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
          scope="col"
        >
          Tipo de Cuenta
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          SomeId
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          A title
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Entry
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`ViewAccountsDashbard Should display the error message 1`] = `
<div>
  <span>
    Something went wrong
  </span>
</div>
`;
